src/custom-element.js

```diff
@@ -489,6 +489,46 @@ function createBaseCustomElementClass(win) {
       ));
     }
 
+    /**
+     * @return {!Promise}
+     * @final
+     */
+    whenLoaded() {
+      return this.signals_.whenSignal(CommonSignals.LOAD_END);
+    }
+
+    /**
+     * Ensure that element is eagerly loaded.
+     *
+     * @param {number=} opt_parentPriority
+     * @return {!Promise}
+     * @final
+     */
+    ensureLoaded(opt_parentPriority) {
+      return this.whenBuilt().then(() => {
+        const resource = this.getResource_();
+        if (this.signals_.get(CommonSignals.LOAD_END)) {
+          return this.whenLoaded();
```

Comment:
Just `return`? We already know it's resolved, right?

Reply:
This is because of the rejection. The `whenLoaded()` will return a rejected promise in that case.

Reply:
Make a note about this.

Reply:
Actually. You were right. This should be a `return` and for only a successful load completion. I fixed this and added more tests.